pilot/pkg/networking/util/util.go

```diff
@@ -71,6 +71,18 @@ const (
 	IstioMetadataKey = "istio"
 	// The range of LoadBalancingWeight is [1, 128]
 	maxLoadBalancingWeight = 128
+
+	// EnvoyTransportSocketMetadataKey is the key under which metadata is added to an endpoint
+	// which determines the endpoint level transport socket configuration.
+	EnvoyTransportSocketMetadataKey = "envoy.transport_socket"
```

Comment:
still under review, but the impl is `https://github.com/envoyproxy/envoy/pull/8354/files#diff-37270bd85c277c4837361a4b59f89b23R65`.

need address before the merge.